### ISSUE TITLE
Add an AuthUI.reauthenticate() function

### ISSUE THREAD

**Issue by samtstern** [firebase/FirebaseUI-Android#563]
There are some Firebase Auth actions that require a recent (~5mins) sign-in to succeed:

  * Delete account (#478)
  * Change email
  * Etc

We should offer a simple method to launch into a re-authentication flow that allows the user to sign in choosing from any of their linked providers.

Considerations:

  * This would likely have to be a method that returns an `Intent` similar to the sign-in intent
  * The developer will probably want to provide a reason for the re-authentication, maybe passed in
  * This would be similar to the normal sign-in flow with a few alterations:
    * Account creation disabled
    * Account picker screen lists only methods the user has already linked to the `FirebaseUser`

**Comment by amandle** [comment]
Unless there are objections I'm going to start taking a crack at this.

**Comment by samtstern** [comment]
Go for it, thanks!

**Comment by nikahmadz** [comment]
May I know , what is the current status of this implementation?

**Comment by samtstern** [comment]
@nikahmadz this is indefinitely on hold ... @amandle left the project and we are focusing on some other priorities for now.

**Comment by maillouxc** [comment]
FWIW, I'd really love to see this implemented. If I'm reading the comments here correctly, it seems like the work was already done for Android, but never released because it wasn't available on the other platforms. Is it possible to just push the release out for android?  I would implement the other devices myself if I had the time, but it seems like a shame to have this stuff ready to go but just waiting on other implementations.

**Comment by TeunVR** [comment]
This is really a necessary feature. 
I am currently using the unreleased iOS FUIAccountSettingsViewController, and it seems to be working for me.
But it should be officially released and also available for Android.
I am a bit sad that i chose for FirebaseUI-Auth now that i am running into this issue :(

**Comment by jackz314** [comment]
It's been more than 2 years from its initial progress and more than a year from the last update, did you guys abandon this key feature?

**Comment by Heldenkrieger01** [comment]
When can we expect the release of this necessary feature? @samtstern

**Comment by katowulf** [comment]
Closing this conversation to prevent more "ETA requested" responses (which actually take time from feature work). 

Latest status update: #issuecomment-367736441

Firebase team will update here and potentially unlock thread if/when there is an update.

**Comment by demolaf** [comment]
This should be resolved in this PR https://github.com/firebase/FirebaseUI-Android/pull/2332, feel free to open a new issue if you come across any bugs.
